rust_src/src/keymap.rs

```diff
@@ -51,6 +53,90 @@ pub extern "C" fn set_where_is_cache_keymaps(val: Lisp_Object) {
     }
 }
 
+/// Check that OBJECT is a keymap (after dereferencing through any
+/// symbols).  If it is, return it.
+///
+/// If AUTOLOAD and if OBJECT is a symbol whose function value
+/// is an autoload form, do the autoload and try again.
+/// If AUTOLOAD, callers must assume GC is possible.
+///
+/// `ERROR_IF_NOT_KEYMAP` controls how we respond if OBJECT isn't a keymap.
+/// If `ERROR_IF_NOT_KEYMAP`, signal an error; otherwise,
+/// just return Qnil.
+///
+/// Note that most of the time, we don't want to pursue autoloads.
+/// Functions like `Faccessible_keymaps` which scan entire keymap trees
+/// shouldn't load every autoloaded keymap.  I'm not sure about this,
+/// but it seems to me that only `read_key_sequence`, `Flookup_key`, and
+/// `Fdefine_key` should cause keymaps to be autoloaded.
+///
+/// This function can GC when AUTOLOAD is true, because it calls
+/// `Fautoload_do_load` which can GC.
+#[no_mangle]
+pub extern "C" fn get_keymap(
+    object: Lisp_Object,
+    error_if_not_keymap: bool,
+    autoload: bool,
+) -> Lisp_Object {
+    let object = LispObject::from_raw(object);
+    let mut autoload_retry;
+    loop {
```

Comment:
This is a while loop based on the `autoload_rety` variable.

Reply:
I initially wanted to use a `do while` loop but learned that Rust doesn't have one. Switched to this one. But I agree maybe a `while` loop is better.